### PR TITLE
New INI setting live_query_max_execution_time to allow configuration of max execution time

### PR DESCRIPTION
**Pull request by tsteur**
cc @mattab 

Users can now optionally configure a max_execution_time for live queries to prevent a live query from running for a very long time eg when looking at a year with a segment that doesn't find any visitors.

Added as many tests as I could. Noticed the MySQL version on Travis doesn't support this feature yet otherwise could have tested it there further.

### CODE REVIEW

File: plugins/Live/Model.php [hunk 1]
```diff
@@ -112,6 +115,84 @@ public function queryLogVisits($idSite, $period, $date, $segment, $offset, $limi
...
+                 * @ignore
+                 * @internal
+                 */
+                Piwik::postEvent('Live.queryMaxExecutionTimeExceeded', array($params));
+                throw new MaxExecutionTimeExceededException($message);
+            }
+            throw $e;
+        }
+        return $visits;
+    }
```
Comment (diosmosis) [inline plugins/Live/Model.php]: This code seems important. Would it be possible to unit/integration test this code w/ a mock Db object returned by `Db::getReader()`?

**Comment by mattab** [comment]
LGTM :+1: (Benaka left a comment above)

**Comment by Findus23** [comment]
Unfortunately this syntax isn't supported in MariaDB which has another synax to set the maximum execution time for a query:
https://mariadb.com/kb/en/library/aborting-statements/

**Comment by tsteur** [comment]
thanks @Findus23 I will mention in the config that it likely won't work on MariaDB

**Comment by tsteur** [comment]
Added some tests @diosmosis

**Pull request merged**